pkg/proxy/proxy.go

```diff
@@ -437,18 +437,18 @@ func (p *Proxy) ReinstallRoutingRules() error {
 		if err := removeFromEgressProxyRoutesIPv6(); err != nil {
 			return err
 		}
-		if !option.Config.EnableIPSec || option.Config.TunnelingEnabled() {
-			if err := removeFromIngressProxyRoutesIPv6(); err != nil {
-				return err
-			}
-		} else {
+		if (option.Config.EnableEnvoyConfig || option.Config.EnableIPSec) && !option.Config.TunnelingEnabled() {
 			ipv6, err := getCiliumNetIPv6()
 			if err != nil {
 				return err
 			}
 			if err := installFromProxyRoutesIPv6(ipv6, defaults.HostDevice); err != nil {
 				return err
```

Comment:
Just to clarify - this means that we now also install the `from-proxy` routes when Envoy is enabled and routing-mode is `native`. But the PR & patch description talk about installing them for `tunnel` routing-mode ...?

Reply:
good catch, I have updated the commit message

Reply:
:pray: I updated the PR title & description accordingly, a bit less entropy in the world.